File: src/main/java/ru/nucodelabs/geo/ves/calc/inverse/inverse_functions/SquaresDiff.java
package ru.nucodelabs.geo.ves.calc.inverse.inverse_functions;

import java.util.List;
import java.util.function.BiFunction;

import static java.lang.StrictMath.sqrt;

public class SquaresDiff implements BiFunction<List<Double>, List<Double>, Double> {
    @Override
    public Double apply(List<Double> solvedResistance, List<Double> experimentalResistance) {
        double functionValue = 0;
        for (int i = 0; i < solvedResistance.size(); i++) {
            functionValue += Math.pow(solvedResistance.get(i) - experimentalResistance.get(i), 2);
        }
        return sqrt(functionValue) / solvedResistance.size();
    }
}

File: src/main/java/ru/nucodelabs/geo/ves/calc/inverse/inverse_functions/SquaresDiff.kt
package ru.nucodelabs.geo.ves.calc.inverse.inverse_functions

class SquaresDiff : (List<Double>, List<Double>) -> Double {
    override fun invoke(solvedResistance: List<Double>, experimentalResistance: List<Double>): Double {
        var functionValue = 0.0
        for (i in solvedResistance.indices) {
            functionValue += (solvedResistance[i] - experimentalResistance[i]).let { it * it }
        }
        return kotlin.math.sqrt(functionValue) / solvedResistance.size
    }
}
